content/en/docs/tasks/administer-cluster/nodelocaldns.md

```diff
@@ -49,12 +49,13 @@ This is the path followed by DNS Queries after NodeLocal DNSCache is enabled:
 {{< figure src="/images/docs/nodelocaldns.svg" alt="NodeLocal DNSCache flow" title="Nodelocal DNSCache flow" caption="This image shows how NodeLocal DNSCache handles DNS queries." >}}
 
 ## Configuration
-{{< note >}} The local listen IP address for NodeLocal DNSCache can be any IP in the 169.254.20.0/16 space or any other IP address that can be guaranteed to not collide with any existing IP. This document uses 169.254.20.10 as an example.
+{{< note >}} The local listen IP address for NodeLocal DNSCache can be any address that can be guaranteed to not collide with any existing IP in your cluster. It's recommended to use an address with a local scope, per example, from the link-local range 169.254.0.0/16 for IPv4 or from the Unique Local Address range in IPv6 fd00::/8.
 {{< /note >}}
 
 This feature can be enabled using the following steps:
 
 * Prepare a manifest similar to the sample [`nodelocaldns.yaml`](https://github.com/kubernetes/kubernetes/blob/master/cluster/addons/dns/nodelocaldns/nodelocaldns.yaml) and save it as `nodelocaldns.yaml.`
+* If using IPv6, the CoreDNS configuration file need to enclose all the IPv6 addresses into square brackets if used in IP:Port format. With current example manifest, this will require to modify [the configuration line](https://github.com/kubernetes/kubernetes/blob/b2ecd1b3a3192fbbe2b9e348e095326f51dc43dd/cluster/addons/dns/nodelocaldns/nodelocaldns.yaml#L70 like this `health [__PILLAR__LOCAL__DNS__]:8080`
```

Comment:
not sure I understand this. User needs to populate only one IP as replacement for "__PILLAR__LOCAL__DNS__". They will use 2 listen IPs - one local IP, one kube-dns IP(if using non-IPVS mode).

Reply:
yeah, and that is still the same, the difference with IPv6 is that in one of the places, in this case *health* ,you need to enclose the IP into square bracket, because it requires an ip:port
If you don' t enclose it coredns will fail to start with the error "invalid address to many colons",(it seems they are using the `net.SplitHostPort`) it is not possible to use ` fd00:8:8:8:8::8:8080`
```
        }                                                                                                                                                                              
        prometheus :9253                                                                                                                                                               
        health [fd00:8:8:8:8::8]:8080     <-------------------                                                                                                                                             
        }                                                                                                                                                                              
    in-addr.arpa:53 {                                                                                                                                                                  
        errors                                                                                                                                                                         
        cache 30                                                                                                                                                                       
        reload                                                                                                                                                                         
        loop                                                                                                                                                                           
        bind fd00:8:8:8:8::8 fd00:10:96::1                                                                                                                                             
        forward . __PILLAR__CLUSTER__DNS__ {                                                                                                                                           
                force_tcp                                                                                                                                                              
        }                                                                                                                                                                              
        prometheus :9253                                                                                                                                                               
        }             
```

Reply:
ok, that makes sense.